source/pastemyst/db/db.d

```diff
@@ -114,6 +112,24 @@ public Nullable!R findOneById(R, T)(T id) @safe
     return findOne!R(["_id": id]);
 }
 
+/++
+ + Removes one item from the DB, using the specified query
+ +/
+public void removeOne(R, T)(T query) @safe
+{
+    MongoCollection collection = mongo[getCollectionName!R()];
+    collection.remove(query);
+}
+
+/++
+ + Removes an existing object, with the specified id
+ +/
+public void removeOneById(R, T)(T id) @safe
+{
+    if (!findOneById!R(id).isNull)
+        removeOne!R(["_id": id]);
```

Comment:
These functions are a bit "lower level" so they should've check if an item exists or not, just remove it. Also, I'm not sure if mongo would complain if you try to delete an item it could't find even.